File: GateOne/oldSnacks/Creditcard.py
def get_card_type(card_number):
    first_digit = card_number[0]
    second_digit = card_number[1]

    if first_digit == '4':
        return "Visacard"
    elif first_digit == '5':
        return "Mastercard"
    elif first_digit == '6':
        return "Discover card"
    elif first_digit == '3' and second_digit == '7':
        return "American Express card"
    else:
        return "Unknown"

def validate_card_number(card_number):
    collecting_number = len(card_number)
    if 13 <= collecting_number or collecting_number <= 16:
        sum_of_doubled_second_digit = 0
        sum_total = 0
        for i in range(0, collecting_number, 2):
            doubled_second_digit = int(card_number[i]) * 2
            sum_of_doubled_second_digit += doubled_second_digit
            if doubled_second_digit >= 10:
                while doubled_second_digit != 0:
                    result = doubled_second_digit % 10
                    sum_total += result
                    doubled_second_digit //= 10

        return sum_of_doubled_second_digit + sum_total

    return "invalid card number"

def cardMain():
    card_number = input("Enter a credit card number: ")
    sum_total = validate_card_number(card_number)
    if sum_total != "invalid card number":
    	print(f"Credit card type: {get_card_type(card_number)}")
    	print(f"Credit card number: {card_number}")
    	print(f"Credit card digit length: {len(card_number)}")
    	if sum_total % 10 == 0:
    	    print("Credit card validity status: Valid")
    	else:
    	    print("Credit card validity status: Invalid")
    else:
        print("Invalid credit card number. Please check your input.")
if __name__== '__main__':
    cardMain()

File: GateOne/oldSnacks/CreditCard.js
const prompt = require("prompt-sync")()

	let doubledSecondDigit = 0;
	let firstDigit = 0;
	let secondDigit =0;
	let result = 0;
	let sumofSingleDigits = 0;
	let sum = 0;
	let sumofdoubledSecondDigit = 0;
	let sumTotal = 0;


	console.log("Hello, Enter card details to verify: "); 
	let cardNumber = prompt();
	//console.log(typeof(cardNumber))

	let collectingNumber = [];	

	if (cardNumber.length >= 13 && cardNumber.length <= 16) {

	for(let i = 0; i < cardNumber.length; i += 2){	

	collectingNumber[i] = cardNumber.charAt(i) - '0';
	
	doubledSecondDigit = collectingNumber[i] * 2;
	
	sumofdoubledSecondDigit += doubledSecondDigit;
	
	if(doubledSecondDigit >= 10) {
	while(doubledSecondDigit != 0) {
	result = doubledSecondDigit % 10;
	sum = sum + result;
	doubledSecondDigit /= 10;
	}
		} 
	}
	sumTotal = sumofdoubledSecondDigit + sum;

	let cardNumberFirstDigit = cardNumber.charAt(0);
	let cardNumberSecondDigit = cardNumber.charAt(1);

		

	if (cardNumberFirstDigit == '4') console.log("Credit card type: Visacard\n");
	
	else if(cardNumberFirstDigit == '5') console.log("Credit card type: Master card\n");
 	
	else if(cardNumberFirstDigit == '6') console.log("Credit card type: Discover card\n");
	
	else if(cardNumberFirstDigit == '3' && cardNumberSecondDigit == '7') console.log("Credit card type: American express card\n");
	
	console.log("Credit card Number: " + cardNumber + "\n");

	console.log("Credit card Digit length: " + cardNumber.length + "\n");

	if (sumTotal % 10 == 0) console.log("Credit card validity status: Valid");
	else  console.log("Credit card validity status: Invalid"); 
}
	else console.log("Put correct number jare\nWerey wan scam me\n");
		

			
	
	
	
 